fluent/src/bundle.js

```diff
@@ -191,23 +191,88 @@ export default class FluentBundle {
    * @param   {Array}              errors
    * @returns {?string}
    */
-  format(message, args, errors) {
-    // optimize entities which are simple strings with no attributes
-    if (typeof message === "string") {
-      return this._transform(message);
+
+  format(path, args, errors) {
+
+    var parts = path.split(".");
+    var id = parts[0];
+    var message = this._messages.get(id);
+
+    if (!this._messages.has(id)) {
+      errors.push(`Message not found: "${id}"`);
+      return null;
     }
 
     // optimize entities with null values
-    if (message === null || message.value === null) {
+    if (parts.length === 1) {
+      // optimize entities which are simple strings with no attributes
+      if (typeof message === "string") {
+        return this._transform(message);
+      }
+      // optimize simple-string entities with attributes
+      if (typeof message.value === "string") {
+        return this._transform(message.value);
+      }
+
+      if (message.value === null) {
+        return null;
+      }
+
+      return resolve(this, args, message, errors);
+    }
+
+    if (parts.length === 2) {
+      if (message.attrs === null) {
+        errors.push(`Message has no attributes: "${id}"`);
+        return null;
+      }
+
+      let attr = message.attrs[parts[1]];
+
+      if (attr === undefined){
+        errors.push(`No attribute called: "${parts[1]}"`);
+        return null;
+      }
+
+      return resolve(this, args, attr, errors);
+    }
+
+    errors.push(`Invalid path: "${path}"`);
+    return null;
+  }
+
+  compound(id, args, errors) {
+
+    if (!this._messages.has(id)) {
+      errors.push(`Message not found: "${id}"`);
       return null;
     }
 
-    // optimize simple-string entities with attributes
-    if (typeof message.value === "string") {
-      return this._transform(message.value);
+    let message = this._messages.get(id);
+
+    if (message.value !== null) {
+      var message_value = resolve(this, args, message, errors);
+    } else {
+      message_value = null;
+    }
+
+    var compoundShape = {
+      value: message_value,
+      attributes: new Map()
+    };
+
+    if (message.attrs === undefined) {
+      return compoundShape;
+    }
+
+    for (let attr of Object.keys(message.attrs)) {
+      compoundShape.attributes.set(
+        attr,
+        message.attrs[attr]
```

Comment:
You'll need to call `resolve` here to format the attribute to a string.

Reply:
I think I did this right.